modal/functions.py

```diff
@@ -1106,20 +1106,46 @@ def remote(self, *args, **kwargs) -> Awaitable[Any]:  # TODO: Generics/TypeVars
                 f"Invoke this function via its web url '{self._web_url}' or call it locally: {self._function_name}()."
             )
         if self._is_generator:
-            return self._call_generator(args, kwargs)  # type: ignore
-        else:
-            return self._call_function(args, kwargs)
+            raise InvalidError(
+                "A generator function cannot be called with `.remote(...)`. Use `.remote_gen(...)` instead."
+            )
 
-    def call(self, *args, **kwargs) -> Awaitable[Any]:  # TODO: Generics/TypeVars
-        deprecation_warning(date(2018, 8, 16), "`f.call(...)` is deprecated. It has been renamed to `f.remote(...)`")
-        return self.remote(*args, **kwargs)
+        return await self._call_function(args, kwargs)
 
-    def shell(self, *args, **kwargs):
-        # TOOD(erikbern): right now fairly duplicated
+    async def remote_gen(self, *args, **kwargs) -> AsyncGenerator[Any, None]:  # TODO: Generics/TypeVars
+        """
+        Calls the generator remotely, executing it with the given arguments and returning the execution's result.
+        """
+        if self._web_url:
+            raise InvalidError(
+                "A web endpoint function cannot be invoked for remote execution with `.remote`. "
+                f"Invoke this function via its web url '{self._web_url}' or call it locally: {self._function_name}()."
+            )
+
+        if not self._is_generator:
+            raise InvalidError(
+                "A non-generator function cannot be called with `.remote_gen(...)`. Use `.remote(...)` instead."
+            )
+        async for item in self._call_generator(args, kwargs):  # type: ignore
+            yield item
+
+    def call(self, *args, **kwargs) -> Awaitable[Any]:  # TODO: Generics/TypeVars
         if self._is_generator:
-            return self._call_generator(args, kwargs)  # type: ignore
+            deprecation_warning(
+                date(2018, 8, 16), "`f.call(...)` is deprecated. It has been renamed to `f.remote(...)`"
```

Comment:
What's up with the date?

Reply:
This is the existing deprecation warning split in two – I think the git diff isn't matching it well

Reply:
You're right. Weird that the year is 2018.

Reply:
lol yeah that's obviously wrong – will fix.

sorry was confused since i just copy pasted